guide/README.md

```diff
@@ -455,6 +455,69 @@ used to bump the `Mi 4` note up by one octave and then to bump the
 | | | For example: Transposing C-D-E-F by 4 (fifth) will give us G-A-B-C
 | | | To transpose an octave: shift by the mode length (7 in major scales) up or down. |
 
+#### <a name="Set Key"></a>3.2.5 Set Key
+
+| Set Key Example |
+| --- |
+| ![set key](./setkey1.svg "set key example") |
+
+
+The *Set key* block is used to set the key and mode, e.g, C Major. 
+It is used to change both the mode and key of the current scale. (The current scale is used to define the mapping of Solfege to notes and also the number of half steps take by the the Scalar step block.
+
+The example above demonstrates traveling up and down the major
+ scale by moving an octave up from the starting note,
+ do, one note at a time and then back down the same way.
+
+
+| Standard Notation for Set Key Example |
+| --- |
+| ![set key notation](./setkey2.png "Standard Notation Set Key") |
+
+
+
+[RUN LIVE](https://musicblocks.sugarlabs.org/index.html?id=1662103714150464&run=True)
+
+When using the *Set key* block, the mode argument is used to define the pattern of half steps and the key argument is used to define the starting position of the pattern. For example, when mode = "major" and key = "C", the pattern of half steps is 2 2 1 2 2 2 1 and the first note in the scale from which the pattern is applied is "C".
+
+Examples are shown in the following table.
+
+| Key | Mode | Notes Pattern | Modes Pattern |
+| --- | --- | --- | --- |
+| C | Major | C, D, E, F, G, A, B, C | 2 2 1 2 2 2 1 |
+| F# | Major | F#, G#, A#, B, C#, D#, F, F# | 2 2 1 2 2 2 1 |
+| A♭ | Major | Ab, Bb, C, Db, Eb, F, G, Ab | 2 2 1 2 2 2 1 |
+| E | Major | E, F#, G#, A, B, C#, D#, E | 2 2 1 2 2 2 1 |
+| A# | Major | Bb, C, D, Eb, F, G, A, Bb | 2 2 1 2 2 2 1 |
+| --- | --- | --- | --- |
+| C | Dorian | C, D, Eb, F, G, A, Bb, C | 2 1 2 2 2 1 2 |
+| F# | Dorian | F#, G#, A, B, C#, D#, E, F# | 2 1 2 2 2 1 2 |
+| A♭ | Dorian | G#, A#, B, C#, D#, F, F#, G# | 2 1 2 2 2 1 2 |
+| E | Dorian | E, F#, G, A, B, C#, D, E | 2 1 2 2 2 1 2 |
+| A# | Dorian | Bb, C, Db, Eb, F, G, Ab, Bb | 2 1 2 2 2 1 2 |
+| --- | --- | --- | --- |
+| C | Phrygian | C, Db, Eb, F, G, Ab, Bb, C | 1 2 2 2 1 2 2 |
+| F# | Phrygian | F#, G, A, B, C#, D, E, F# | 1 2 2 2 1 2 2 |
+| A♭ | Phrygian | G#, A, B, C#, D#, E, F#, G# | 1 2 2 2 1 2 2 |
+| E | Phrygian | E, F, G, A, B, C, D, E | 1 2 2 2 1 2 2 |
+| A# | Phrygian | Bb, Cb, Db, Eb, F, Gb, Ab, Bb | 1 2 2 2 1 2 2 |
+
+**Set Key & Scalar Step**
+
+The set key block helps us in selecting the scale and half-steps goes over all the scales
```

Comment:
The *Set key* block is used to select a subset of notes in the given temperament. (By default, Music Blocks uses equal temperament 12. The key and mode determine which of these notes will be used.)

Reply:
The _Set key_ block is used to select a subset of notes in the given temperament. (By default, Music Blocks uses equal temperament 12. The key and mode determine which of these notes will be used.)

The _Set key_ block helps us in selecting the scale and half-steps goes over all the scales
while scalar step on the other hand will walk only through those nodes that are selected 
for the representation of the music.

**Like this, or just** 

The Set key block is used to select a subset of notes in the given temperament. (By default, Music Blocks uses equal temperament 12. The key and mode determine which of these notes will be used.)

Reply:
I think the 2nd phrasing is clearer.